src/main/scala/gitbucket/core/controller/RepositoryViewerController.scala

```diff
@@ -670,12 +670,14 @@ trait RepositoryViewerControllerBase extends ControllerBase {
 
   private def archiveRepository(name: String, suffix: String, repository: RepositoryService.RepositoryInfo): Unit = {
     val revision = name.stripSuffix(suffix)
-
-    val filename = repository.name + "-" +
-      (if(revision.length == 40) revision.substring(0, 10) else revision).replace('/', '-') + suffix
-
+    
     using(Git.open(getRepositoryDir(repository.owner, repository.name))){ git =>
       val revCommit = JGitUtil.getRevCommitFromId(git, git.getRepository.resolve(revision))
+      
+      val sha1 = git.getRepository.resolve(revision).getName()
```

Comment:
Thank you for your comment.

After some trial, I found my solution.

I'm not familliar with scala, I welcome any suggestion.